hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanActionExecutor.java

```diff
@@ -58,8 +59,34 @@ public CleanPlanActionExecutor(HoodieEngineContext context,
     this.extraMetadata = extraMetadata;
   }
 
-  protected Option<HoodieCleanerPlan> createCleanerPlan() {
-    return execute();
+  private int getCommitInfo() {
+    Option<HoodieInstant> lastCleanInstant = table.getActiveTimeline().getCleanerTimeline().filterCompletedInstants().lastInstant();
+    HoodieTimeline commitTimeline = table.getActiveTimeline().getCommitTimeline().filterCompletedInstants();
+
+    String latestCleanTs;
+    int commitsSinceLastCleaning = 0;
+    if (lastCleanInstant.isPresent()) {
+      latestCleanTs = lastCleanInstant.get().getTimestamp();
+      commitsSinceLastCleaning = commitTimeline.findInstantsAfter(latestCleanTs).countInstants();
+    } else {
+      String firstCommitTs = commitTimeline.firstInstant().get().getTimestamp();
+      commitsSinceLastCleaning = commitTimeline.findInstantsAfterOrEquals(firstCommitTs, Integer.MAX_VALUE).countInstants();
```

Comment:
is n't this same as `commitTimeline.countInstants()`?

Reply:
right.